citation/graphviz/data.py

```diff
@@ -144,27 +150,28 @@ def generate_aggregated_distribution_data(filter_criteria, classifier, name):
             if date_published is not None:
                 years_list.append(date_published)
                 bucket = availability if is_archived else non_availability
-                bucket.append(date_published)
+                bucket[date_published] += 1
 
         distribution_data = []
-        availability_counter = Counter(availability)
-        non_availability_counter = Counter(non_availability)
         count = len(years_list)
         for year in set(years_list):
-            present = availability_counter[year] * 100 / count
-            absent = non_availability_counter[year] * 100 / count
+            present = availability[year] * 100 / count
+            absent = non_availability[year] * 100 / count
             total = present + absent
-            distribution_data.append(
-                {'relation': classifier, 'name': name, 'date': year,
-                 'Code Available': availability_counter[year],
-                 'Code Not Available': non_availability_counter[year],
-                 'Code Available Per': present * 100 / total,
-                 'Code Not Available Per': absent * 100 / total})
+            distribution_data.append({
+                'relation': classifier, 'name': name, 'date': year,
+                'Code Available': availability[year],
+                'Code Not Available': non_availability[year],
+                'Code Available Per': present * 100 / total,
+                'Code Not Available Per': absent * 100 / total
+            })
 
         return distribution_data
 
 
-def generate_aggregated_code_archived_platform_data(filter_criteria = {}):
+def generate_aggregated_code_archived_platform_data(filter_criteria=None):
```

Comment:
just as a warning @dhruvilpatel you should never use mutable data structures as default keyword arguments, see https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html for more details.